sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md

```diff
@@ -0,0 +1,255 @@
+                    if ("Name".equals(key)) {
+                        if (DocumentFieldType.STRING == formField.getType()) {
+                            String name = formField.getValueString();
+                            System.out.printf("Name: %s, confidence: %.2fs%n",
+                                name, formField.getConfidence());
+                        }
+                    }
+                    if ("Quantity".equals(key)) {
+                        if (DocumentFieldType.FLOAT == formField.getType()) {
+                            Float quantity = formField.getValueFloat();
+                            System.out.printf("Quantity: %f, confidence: %.2f%n",
+                                quantity, formField.getConfidence());
+                        }
+                    }
+                }));
+        }
+    }
+}
+```
+
+#### Build a custom document analysis model
+In 3.X, creating a custom model required specifying `useTrainingLabels` to indicate if the model creation was using the custom labeling or not with the `beginTraining` method.
+In 4.X, we introduced the new general document model (prebuilt-document) consisting of key-value pair extraction, it is no longer necessary to train a custom model to extract key-value pairs. Thus, we removed the option to train custom models without labels. Hence, with 4.X, we can build a custom document analysis model using the `beginBuildModel` method as shown in the example below:
+<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L293-L315 -->
+```java
+// Build custom document analysis model
+String trainingFilesUrl = "{SAS_URL_of_your_container_in_blob_storage}";
+// The shared access signature (SAS) Url of your Azure Blob Storage container with your forms.
+SyncPoller<DocumentOperationResult, DocumentModel> buildOperationPoller =
+    documentModelAdminClient.beginBuildModel(trainingFilesUrl,
+        "my-build-model",
+        new BuildModelOptions().setDescription("model desc"),
+        Context.NONE);
+
+DocumentModel documentModel = buildOperationPoller.getFinalResult();
+
+// Model Info
+System.out.printf("Model ID: %s%n", documentModel.getModelId());
+System.out.printf("Model Description: %s%n", documentModel.getDescription());
+System.out.printf("Model created on: %s%n%n", documentModel.getCreatedOn());
+documentModel.getDocTypes().forEach((key, docTypeInfo) -> {
+    System.out.printf("Document type: %s%n", key);
+    docTypeInfo.getFieldSchema().forEach((name, documentFieldSchema) -> {
+        System.out.printf("Document field: %s%n", name);
+        System.out.printf("Document field type: %s%n", documentFieldSchema.getType().toString());
+        System.out.printf("Document field confidence: %.2f%n", docTypeInfo.getFieldConfidence().get(name));
+    });
+});
+```
+
+### Manage models
+In 3.X, listing models returned only the custom trained models using the `listCustomModel` method.
```

Comment:
we should also call out how these methods now only return models that are ready to be consumed, while before models in whatever state where returned